### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by JuantAldea**
This one has been quite a ride, got it down to 59952 bytes...

### CODE REVIEW

**Review by egzumer — COMMENTED** [review]
There are bugs in here. If you disable FM radio in Makefile there is a build error, also if suppressed then the F key doesn't work.

**Review by egzumer — COMMENTED** [review]
also in FM mode, if you start entering channel number or frequency and try to lock the keyboard, it gives out invalid key sound but still locks

**Review by egzumer — COMMENTED** [review]
this brakes the input box for FM radio channel input, you get "1-------"

**Comment by JuantAldea** [comment]
I'll look into it

**Review by egzumer — COMMENTED** [review]
Now on main screen when you start entering frequency or channel number, and you hit F you get reject beep, but the F still is activated.
:)

**Comment by egzumer** [comment]
Maybe that would help :)
[ConsoleApplication1.zip](https://github.com/egzumer/uv-k5-firmware-custom/files/13795130/ConsoleApplication1.zip)

File: app/fm.c
Comment (egzumer) [inline app/fm.c:1]: This change is confusing, doesn't seem to have anything to do with PTT, also this function is only used in one place

File: app/generic.c [hunk 1]
```diff
@@ -39,69 +39,56 @@
...
-			#ifdef ENABLE_VOICE
-				if (!gWasFKeyPressed)
-					gAnotherVoiceID = VOICE_ID_CANCEL;
-			#endif
-
-			gUpdateStatus = true;
+		// short press -> pressing
+#ifdef ENABLE_FMRADIO
+		if (gScreenToDisplay == DISPLAY_FM) {
+			return FM_Key_PTT();
```
Comment (egzumer) [inline app/generic.c:60]: A bit confusing... return result from function returning void?

File: app/fm.c
Comment (JuantAldea) [reply app/fm.c:1]: Oh yeah, Freudian slip...

**Comment by JuantAldea** [comment]
> Now on main screen when you start entering frequency or channel number, and you hit F you get reject beep, but the F still is activated. :)

Uhm, I'm not managing to reproduce this one. But I have noticed that all F-key action is supposed to be inhibited while the input is active.

**Review by egzumer — COMMENTED** [review]
I wouldn't touch that code. It is messed up and touching anything here breaks things, and looking for bugs is a pain in the ass.

If you check out on this commit it is horribly broken, if you input 1 tone it transmits it repeatedly multiple times, something further down the line fixes it, but I don't even want to investigate what.

There are other problems also.
1. The PTT doesn't close the input box if it is empty.
2. If you transmit 10 tones, enter 10 other tones again, delete 9 tones without transmitting, hit PTT, it will transmit 10 tones not the 1 that is left.
3. Also "-" characters act weird in the process.

**Comment by JuantAldea** [comment]
Yeah, all this is very, very fragile... to particular commit are you referring? Also... how do you delete a single tone? Pressing EXIT deletes the whole string.

**Comment by egzumer** [comment]
Over the comment there is a link, view reviewed changes.

Side button 1.

**Comment by JuantAldea** [comment]
Not yet. Pushed to the wrong branch :D

**Comment by JuantAldea** [comment]
On a side note, I have noticed that sometimes there's a conflict printing into the central line of the display.
Disregard this:
> Not yet. Pushed to the wrong branch :D
Above commit seems good.

**Comment by egzumer** [comment]
> On a side note, I have noticed that sometimes there's a conflict printing into the central line of the display.

Never seen it. What happens?

**Comment by egzumer** [comment]
One of the previous changes broke STE when roger is used. There is also #353. I rolled back the changes.
My main goal is to keep this firmware in a working order, there are to many issues. Lets not make changes that are hard to verify.

**Comment by JuantAldea** [comment]
Fair enough

**Comment by JuantAldea** [comment]
> > On a side note, I have noticed that sometimes there's a conflict printing into the central line of the display.
> 
> Never seen it. What happens?

So I have managed to  characterize it. Send a DTMF string, a long one. With another radio, start TX. The RSSI bar will appear in the radio has sent the string,, fighting with the DTMF string that is showing in the central line.

**Comment by JuantAldea** [comment]
https://github.com/egzumer/uv-k5-firmware-custom/assets/1664307/0700a737-99f8-46ed-b936-dbc020602f0d

**Comment by egzumer** [comment]
fixed ac8c51d6052cf280e4683ec2efe160fc47c72ba6

**Comment by egzumer** [comment]
Sorry to point out, but another one #366. I appreciate help, hope you won't get discouraged, bugs happen, just be mindful :)

**Pull request closed without merging**